standard/classes.md

```diff
@@ -1565,7 +1565,7 @@ A constant declaration that declares multiple constants is equivalent to multipl
 >
 > *end example*
 
-Constants are permitted to depend on other constants within the same program as long as the dependencies are not of a circular nature. The compiler automatically arranges to evaluate the constant declarations in the appropriate order.
+Constants are permitted to depend on other constants within the same program as long as the dependencies are not of a circular nature. A compiler is allowed to evaluate the constant declarations in an appropriate order.
```

Comment:
(I can't see how to re-surface @jskeet's comment in “Files Changed”, I've “unresolved” the conversation so its visible in “Conversation”.)

@jskeet asked whether this should be a language requirement.

I think a lot of the original source material *described* what “the compiler” (it was the one and only) did and was never converted into language requirements.

“A compiler is allowed to evaluate the constant declarations in an appropriate order.” implies that it is not required to do so, which is clearly not true. I think any appropriate order was probably meant?

```suggestion
Constants are permitted to depend on other constants within the same program as long as the dependencies are not of a circular nature. A compiler must evaluate the constant declarations in an appropriate order.
```

But are we saying too much? Do compiler writers require this much direction to order the dependencies? Here I might offer:

```suggestion
Constants are permitted to depend on other constants within the same program as long as the dependencies are not of a circular nature.
```

If it is felt that compiler writers need more help to figure this out we could go with something along the lines of:

```suggestion
Constants are permitted to depend on other constants within the same program as long as the dependencies are not of a circular nature. A compiler must evaluate constant declarations in an order which respects the dependencies.
```

or maybe:

```suggestion
Constants are permitted to depend on other constants within the same program as long as the dependencies are not of a circular nature. Constant declarations shall be evaluated in any order which respects any dependencies.
```

I personally lean towards the single sentence option, compiler writers can figure it out.